apps/member-profile/app/routes/_profile.profile.emails.tsx

```diff
@@ -64,40 +140,53 @@ function EmailAddressSection() {
         school, personal, work), please add them here. Your primary email is the
         email where you will receive all ColorStack communications.
       </ProfileDescription>
+      <RemixForm className="form" method="post">
+        <ul className="flex flex-col gap-2">
+          {emails.map((email) => {
+            return (
+              <li
+                className={cx(
+                  'flex items-center justify-between rounded-lg border border-solid p-2',
+                  email.primary ? 'border-gold bg-gold-100' : 'border-gray-200'
+                )}
+                key={email.email}
+              >
+                <Text>{email.email}</Text>
+
+                {email.primary && (
+                  <Text
+                    className="font-medium uppercase text-gold"
+                    variant="sm"
+                  >
+                    Primary
+                  </Text>
+                )}
+              </li>
+            );
+          })}
+        </ul>
+        <AllowEmailShareField
+          defaultValue={student.allowEmailShare}
+          error={errors.AllowEmailShare}
+          name={AllowEmailShare}
+        />
+        <Button.Group>
+          <Button onClick={onAddEmail} size="small" variant="secondary">
+            <Plus /> Add Email
+          </Button>
 
-      <ul className="flex flex-col gap-2">
-        {emails.map((email) => {
-          return (
-            <li
-              className={cx(
-                'flex items-center justify-between rounded-lg border border-solid p-2',
-                email.primary ? 'border-gold bg-gold-100' : 'border-gray-200'
-              )}
-              key={email.email}
-            >
-              <Text>{email.email}</Text>
-
-              {email.primary && (
-                <Text className="text-gold font-medium uppercase" variant="sm">
-                  Primary
-                </Text>
-              )}
-            </li>
-          );
-        })}
-      </ul>
-
-      <Button.Group>
-        <Button onClick={onAddEmail} size="small" variant="secondary">
-          <Plus /> Add Email
-        </Button>
-
-        {emails.length > 1 && (
-          <Button color="primary" onClick={onChangePrimaryEmail} size="small">
-            <Edit /> Change Primary
+          {emails.length > 1 && (
+            <Button color="primary" onClick={onChangePrimaryEmail} size="small">
+              <Edit /> Change Primary
+            </Button>
+          )}
+        </Button.Group>
+        <Button.Group>
+          <Button loading={submitting} type="submit">
+            Save
```

Comment:
In this particular case (a 1-field form with checkbox), I think it might more sense if we omitted any "Save" button altogether. What if we just save the update as soon as the checkbox is checked?

To do that, we can do something like:

```
const submit = useSubmit();

<RemixForm onChange={(e) => submit(e.currentTarget)}>
  // ...
</RemixForm>
```

Reply:
Ok, got it working without the "Save" button. It was a simple change after all 😄